### PR TITLE
add test of unions

### PR DESCRIPTION
**Pull request by changlong-liu**
# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/http/unions/main.cadl [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+""")
+@route("/int-array")
+@post
+op sendIntArray(@body input: WrapperModel1): OkResponse;
+
+@scenario
+@scenarioDoc("""
+This test is testing sending Installation1 in named union property.
+```json
+{ "namedUnion": { "name": "installation1", "prop1": 1 } }
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/unions/main.cadl:65]: Is there any doc about the named Union? E.g. why discriminator be "name", not "kind" or "type"?

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/unions/main.cadl:65]: Thanks Changlong. I see, name/prop1 is from "Installation1" model.

LGTM.

Comment (changlong-liu) [reply packages/cadl-ranch-specs/http/unions/main.cadl:65]: yes, Weidong. Almost all the same, excepts there is a name for named union.

**Review by nguerrera — APPROVED** [review]

File: packages/cadl-ranch-specs/http/unions/main.cadl [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+@scenarioService("/unions")
+namespace Unions;
+
+@doc("This is a base model.")
+model BaseInstallation {
+  name: string;
+}
+
+@doc("The first one of the unioned model type.")
+model Installation1 extends BaseInstallation {
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/unions/main.cadl:17]: Maybe `Model` is better than `Installation`?

File: packages/cadl-ranch-specs/http/unions/main.cadl [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  prop2: int32;
+}
+
+union MyNamedUnion {
+  one: Installation1,
+  two: Installation2,
+}
+
+model WrapperModel1 {
+  simpleUnion: int32 | int32[];
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/unions/main.cadl:32]: `ModelWithSimpleUnionProperty`, and `ModelWithNamedUnionProperty`

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 4]
```diff
@@ -1735,3 +1735,43 @@ Expected input body:
...
+
+```json
+{ "simpleUnion": [1, 2] }
+```
+
+### Unions_sendInstallation1
+
+- Endpoint: `post /unions/installation1`
+
+This test is testing sending Installation1 in named union property.
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:1763]: what is Installation1? We really want the test scenario names to clearly convey what the test is for just through the name. Maybe `sendFirstUnionValue` and `sendSecondUnionValue`?

Comment (changlong-liu) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md]: Thanks for the suggestions @iscai-msft , I updated for them.

**Pull request merged**